File: server/internal/domain/daily_bill.go
package domain

import "time"

type DailyBill struct {
	ID             uint64    `gorm:"primaryKey" json:"id"`
	BillDate       time.Time `gorm:"not null;uniqueIndex:idx_bill_unique" json:"bill_date"`
	UserID         uint64    `gorm:"not null;uniqueIndex:idx_bill_unique" json:"user_id"`
	ModelName      string    `gorm:"size:128;not null;uniqueIndex:idx_bill_unique" json:"model_name"`
	RequestCount   int       `gorm:"not null;default:0" json:"request_count"`
	TokenCount     int       `gorm:"not null;default:0" json:"token_count"`
	QuotaConsumed  int64     `gorm:"not null;default:0" json:"quota_consumed"`
	QuotaRecharged int64     `gorm:"not null;default:0" json:"quota_recharged"`
	CreatedAt      time.Time `json:"created_at"`
	UpdatedAt      time.Time `json:"updated_at"`
}

func (DailyBill) TableName() string { return "daily_bills" }

File: admin/src/api/dailyBill.ts
import { client } from './client'
import type { ApiResponse, PagedResponse } from '../types/api'

export interface DailyBill {
  id: number
  bill_date: string
  user_id: number
  model_name: string
  request_count: number
  token_count: number
  quota_consumed: number
  quota_recharged: number
  created_at: string
  updated_at: string
}

export interface MonthlyBill {
  month: string
  request_count: number
  token_count: number
  quota_consumed: number
  quota_recharged: number
}

export function listDailyBills(
  page = 1,
  pageSize = 20,
  startDate: string,
  endDate: string,
  userId?: number,
) {
  let url = `/daily-bills?page=${page}&page_size=${pageSize}&start_date=${encodeURIComponent(
    startDate,
  )}&end_date=${encodeURIComponent(endDate)}`
  if (userId) {
    url += `&user_id=${userId}`
  }
  return client.get<ApiResponse<PagedResponse<DailyBill>>, ApiResponse<PagedResponse<DailyBill>>>(url)
}

export function listMonthlyBills(startMonth: string, endMonth: string, userId?: number) {
  let url = `/daily-bills/monthly?start_month=${encodeURIComponent(startMonth)}&end_month=${encodeURIComponent(
    endMonth,
  )}`
  if (userId) {
    url += `&user_id=${userId}`
  }
  return client.get<ApiResponse<PagedResponse<MonthlyBill>>, ApiResponse<PagedResponse<MonthlyBill>>>(url)
}

export function aggregateDailyBills(date: string) {
  return client.post<ApiResponse<unknown>, ApiResponse<unknown>>(
    `/daily-bills/aggregate?date=${encodeURIComponent(date)}`,
  )
}
